modules/currency.js

```diff
@@ -0,0 +1,206 @@
+
+import bidfactory from 'src/bidfactory';
+import { STATUS } from 'src/constants';
+import { ajax } from 'src/ajax';
+import * as utils from 'src/utils';
+import bidmanager from 'src/bidmanager';
+
+$$PREBID_GLOBAL$$.currency = setConfig;
+
+const DEFAULT_CURRENCY_RATE_URL = "http://prebid.aws.rubiconproject.com/latest.json";
+const CURRENCY_RATE_PRECISION = 4;
+
+var bidResponseQueue = [];
+var conversionCache = {};
+var currencyRatesLoaded = false;
+var adServerCurrency = 'USD';
+
+var oldBidResponse;
+
+export var currencySupportEnabled = false;
+export var currencyRates = {};
+
+export function setConfig(config) {
+  var url = DEFAULT_CURRENCY_RATE_URL;
+  if (typeof config.adServerCurrency === 'string') {
+    utils.logInfo('enabling currency support', arguments);
+
+    adServerCurrency = config.adServerCurrency;
+    if (config.conversionRateFile) {
+      utils.logInfo('currency using override conversionRateFile:', config.conversionRateFile);
+      url = config.conversionRateFile;
+    }
+    initCurrency(url);
+  } else {
+    // currency support is disabled, setting defaults
+    utils.logInfo('disabling currency support');
+    resetCurrency();
+  }
+}
+
+function initCurrency(url) {
+  utils.logInfo('Invoking initCurrency', arguments);
+
+  conversionCache = {};
+  currencySupportEnabled = true;
+
+  if(!oldBidResponse) {
```

Comment:
nit: I think the current style is a space after `if` but the linter doesn't seem to be complaining about it being mixed. I would just run a fix on the file.